st2api/st2api/controllers/resource.py

```diff
@@ -182,6 +182,16 @@ def _get_all(self, exclude_fields=None, include_fields=None,
             else:
                 filters['__'.join(v.split('.'))] = filter_value
 
+        if advanced_filters:
```

Comment:
Can we please also add some test cases for this.

Reply:
we absolutely will